### PR TITLE
feat(simple_planning_simulator): add VGR sim model

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

As shown in https://github.com/autowarefoundation/autoware.universe/pull/8504, add conversion from steer angle to handle angle by variable gear ratio to raw_vehicle_cmd_converter.
To simulate this, implement a model that works with handle angle. And add atuation status publishing.

image

1) same VGR coef in converter and sim
raw_vehicle_cmd_converter
```
    convert_steer_cmd_method: "vgr"
    vgr_coef_a: 15.713
    vgr_coef_b: 0.053
    vgr_coef_c: 0.042
    convert_actuation_to_steering_status: true
```

sim
```
    accel_time_delay: 0.1
    accel_time_constant: 0.1
    brake_time_delay: 0.1
    brake_time_constant: 0.1
    accel_map_path: $(find-pkg-share simple_planning_simulator)/data/actuation_cmd_map/accel_map.csv
    brake_map_path: $(find-pkg-share simple_planning_simulator)/data/actuation_cmd_map/brake_map.csv

    convert_accel_cmd: true
    convert_brake_cmd: true
    convert_steer_cmd: true

    convert_steer_cmd_method: "vgr"
    vgr_coef_a: 15.713
    vgr_coef_b: 0.053
    vgr_coef_c: 0.042
    enable_pub_steer: false
```

https://github.com/user-attachments/assets/c17a0926-270d-4931-8232-f4a1ba97748d

2) change sim VGR
```
    convert_steer_cmd_method: "vgr"
    vgr_coef_a: 5.0
    vgr_coef_b: 2.0
    vgr_coef_c: 1.0
    convert_actuation_to_steering_status: true
    enable_pub_steer: false
```

https://github.com/user-attachments/assets/cdf3149f-09f4-4763-928e-058c23058fc6

## Related links

**Parent Issue:**

- Link

## How was this PR tested?

- psim 
- add unit test

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: simulator/simple_planning_simulator/test/test_simple_planning_simulator.cpp [hunk 1]
```diff
@@ -277,10 +298,23 @@ TEST_P(TestSimplePlanningSimulator, TestIdealSteerVel)
   rclcpp::init(0, nullptr);
 
   const auto params = GetParam();
-  const auto command_type = std::get<0>(params);
-  const auto vehicle_model_type = std::get<1>(params);
-
+  // common parameters
+  const auto common_params = get_common_params(params);
+  auto command_type = common_params.first;
+  auto vehicle_model_type = common_params.second;
```
Comment (kosuke55) [inline simulator/simple_planning_simulator/test/test_simple_planning_simulator.cpp:304]:
```suggestion
  const auto command_type = common_params.first;
  const auto vehicle_model_type = common_params.second;
```

File: simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp [hunk 2]
```diff
@@ -454,6 +478,14 @@ void SimplePlanningSimulator::on_timer()
   publish_turn_indicators_report();
   publish_hazard_lights_report();
   publish_tf(current_odometry_);
+
+  if (enable_pub_steer_) {
+    publish_steering(current_steer_);
+  }
+
+  if (vehicle_model_type_ == VehicleModelType::ACTUATION_CMD) {
```
Comment (TakaHoribe) [inline simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp:486]: The `VehicleModelType` should not be used here for extensibility. Please define steering_publish_type in constructor, use it here.

File: simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp [hunk 3]
```diff
@@ -850,6 +882,25 @@ void SimplePlanningSimulator::publish_tf(const Odometry & odometry)
   tf_msg.transforms.emplace_back(std::move(tf));
   pub_tf_->publish(tf_msg);
 }
+
+void SimplePlanningSimulator::publish_actuation_status()
+{
+  if (vehicle_model_type_ != VehicleModelType::ACTUATION_CMD) {
+    return;
+  }
+  const auto model = std::dynamic_pointer_cast<SimModelActuationCmd>(vehicle_model_ptr_);
```
Comment (TakaHoribe) [inline simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp:891]: The use of `dynamic_pointer_cast` should be avoided as much as possible. In this case, you can add a  getActuationStatus() in a base class with returning zero by default.

Comment (kosuke55) [reply simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp]: fixed in https://github.com/autowarefoundation/autoware.universe/pull/8415/commits/ed78398b7705c83460c9089f2aef53019f552e9f

File: simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp (hunk 2 above)
Comment (kosuke55) [reply simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp]: fixed in https://github.com/autowarefoundation/autoware.universe/pull/8415/commits/ed78398b7705c83460c9089f2aef53019f552e9f

File: simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp (hunk 3 above)
Comment (kosuke55) [reply simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp]: >  returning zero by default.

I return not zoro but null

File: simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp (hunk 2 above)
Comment (kosuke55) [reply simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp]: added `virtual bool shouldPublishActuationStatus() const { return false; }`

**Review by TakaHoribe — APPROVED** [review]

**Pull request merged**